Cabal/doc/getting-started.rst

```diff
@@ -0,0 +1,159 @@
+Getting Started with Haskell and Cabal
+======================================
+
+Installing the Haskell toolchain
+--------------------------------
+
+To install the Haskell toolchain follow the `ghcup instructions
+<https://www.haskell.org/ghcup/>`__.
+
+
+Creating a new application
+--------------------------
+
+Let's start by creating a simple Haskell application from scratch where we'll
+learn about a Haskell package's directory structure, how to run the executable,
+and how to add external dependencies.
+
+
+Initializing the application
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Start by creating a ``myfirstapp`` directory to hold the project files:
+
+.. code-block:: console
+
+    $ mkdir myfirstapp && cd myfirstapp
+
+Once you have an empty directory we can initialize our package:
+
+.. code-block:: console
+
+    $ cabal init
+
+This will generate the following files:
+
+.. code-block:: console
+
+    $ ls
+    CHANGELOG.md
+    Main.hs
+    myfirstapp.cabal
+
+
+``Main.hs`` is where your package's code lives. By default ``cabal init``
+creates an executable with the same name as the package ``myfirstapp`` in this
+case, you can instruct ``cabal init`` to generate just a library (with
+``--lib``) or both a library and executable with (``--libandexe``); for the full
+set of options see ``cabal init --help``.
+
+``myfirstapp.cabal`` is Cabal's metadata file which describes your package and
+its dependencies. We'll be updating this file in a little bit when we add an
+external dependency to our package.
+
+.. note:: By default the name of the current directory will be used for the
+	  package name, you can change this with ``-p <package-name>``.
```

Comment:
I think it's better to supply package name explicitly...

Reply:
Done.